### PR TITLE
bpo-32323: urllib.parse.urlsplit() must not lowercase() IPv6 scope value

### PR DESCRIPTION
**Pull request by socketpair**
https://bugs.python.org/issue32323

### CODE REVIEW

**Comment by socketpair** [comment]
@asvetlov

File: Lib/urllib/parse.py [hunk 1]
```diff
@@ -155,9 +155,19 @@ def password(self):
     def hostname(self):
         hostname = self._hostinfo[0]
         if not hostname:
-            hostname = None
-        elif hostname is not None:
+            return None
+        # Scoped IPv6 address may have zone info, which must not be lowercased
+        # like http://[fe80::822a:a8ff:fe49:470c%tESt]:1234/keys
+        if isinstance(hostname, str):
+            (hostname, *zone) = hostname.split('%', 1)
```
Comment (asvetlov) [inline Lib/urllib/parse.py]: Drop parentheses at left of `=`, the module doesn't use this style.

File: Lib/urllib/parse.py [hunk 2]
```diff
@@ -155,9 +155,19 @@ def password(self):
...
             hostname = hostname.lower()
+            if zone:
+                hostname += '%' + zone[0]
+        else:
+            (hostname, *zone) = hostname.split(b'%', 1)
```
Comment (asvetlov) [inline Lib/urllib/parse.py]: Drop parentheses

File: Lib/urllib/parse.py (hunk 1 above)
Comment (asvetlov) [reply Lib/urllib/parse.py]: I recommend considering `hostname.partition('%')` -- it makes code cleaner.

File: Lib/urllib/parse.py (hunk 2 above)
Comment (socketpair) [reply Lib/urllib/parse.py]: fixed

File: Lib/urllib/parse.py (hunk 1 above)
Comment (socketpair) [reply Lib/urllib/parse.py]: fixed

**Review by asvetlov — APPROVED** [review]

**Pull request merged**